### PR TITLE
kusto_cluster_resource - allowed_fqdns, allowed_ip_ranges, outbound_network_access_restricted

### PR DESCRIPTION
**Pull request by liuwuliuyun**
1. Add `allowed_fqdn_list`, `allowed_ip_range_list` and `restrict_outbound_network_access` to kusto cluster resource
2. Update supported sku by https://docs.microsoft.com/en-us/azure/templates/microsoft.kusto/clusters?tabs=bicep#azuresku

### CODE REVIEW

**Comment by liuwuliuyun** [comment]
GOROOT=D:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"D:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto #gosetup
"D:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccKustoCluster_complete\E$ #gosetup
=== RUN   TestAccKustoCluster_complete
=== PAUSE TestAccKustoCluster_complete
=== CONT  TestAccKustoCluster_complete
--- PASS: TestAccKustoCluster_complete (1287.03s)
PASS


Process finished with the exit code 0

**Comment by liuwuliuyun** [comment]
GOROOT=D:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"D:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto #gosetup
"D:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccKustoCluster_basic\E$ #gosetup
=== RUN   TestAccKustoCluster_basic
=== PAUSE TestAccKustoCluster_basic
=== CONT  TestAccKustoCluster_basic
--- PASS: TestAccKustoCluster_basic (1099.87s)
PASS


Process finished with the exit code 0

File: website/docs/r/kusto_cluster.html.markdown [hunk 1]
```diff
@@ -46,6 +46,10 @@ The following arguments are supported:
 
 * `sku` - (Required) A `sku` block as defined below.
 
+* `allowed_fqdn_list` - (Optional) List of allowed FQDNs(Fully Qualified Domain Name) for egress from Cluster.
+
+* `allowed_ip_rangelist` - (Optional) The list of ips in the format of CIDR allowed to connect to the cluster.
```
Comment (liranms) [inline website/docs/r/kusto_cluster.html.markdown:51]: should be `allowed_ip_rangelist_list` (you are missing underscore between range and list)

Comment (liuwuliuyun) [reply website/docs/r/kusto_cluster.html.markdown:51]: oh, let me change that typo

File: internal/services/kusto/kusto_cluster_resource.go [hunk 2]
```diff
@@ -108,6 +128,24 @@ func resourceKustoCluster() *pluginsdk.Resource {
 				},
 			},
 
+			"allowed_fqdn_list": {
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:131]: i don't think we need list here?
```suggestion
			"allowed_fqdn_list": {
```
```suggestion
			"allowed_fqdns": {
```

File: internal/services/kusto/kusto_cluster_resource.go [hunk 3]
```diff
@@ -108,6 +128,24 @@ func resourceKustoCluster() *pluginsdk.Resource {
...
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+
+			"allowed_ip_range_list": {
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:140]:
```suggestion
			"allowed_ip_ranges": {
```

File: internal/services/kusto/kusto_cluster_resource.go [hunk 4]
```diff
@@ -213,6 +251,17 @@ func resourceKustoCluster() *pluginsdk.Resource {
 				Default:  true,
 			},
 
+			"restrict_outbound_network_access": {
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:254]: we word bools as
```suggestion
			"outbound_network_access_restricted": {
```
```suggestion
			"restrict_outbound_network_access": {
```

File: internal/services/kusto/kusto_cluster_resource.go [hunk 5]
```diff
@@ -213,6 +251,17 @@ func resourceKustoCluster() *pluginsdk.Resource {
...
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(kusto.ClusterNetworkAccessFlagDisabled),
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(kusto.ClusterNetworkAccessFlagEnabled),
+						string(kusto.ClusterNetworkAccessFlagDisabled),
+					}, false),
```
Comment (katbyte) [inline internal/services/kusto/kusto_cluster_resource.go:262]: this should be a boolean?
```suggestion
				Type:     pluginsdk.TypeBool,
				Optional: true,
				Default:  false,
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by liuwuliuyun** [comment]
Let me change according to review and resubmit in this week.

**Comment by liuwuliuyun** [comment]
GOROOT=D:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"D:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto #gosetup
"D:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccKustoCluster_basic\E$ #gosetup
=== RUN   TestAccKustoCluster_basic
=== PAUSE TestAccKustoCluster_basic
=== CONT  TestAccKustoCluster_basic
--- PASS: TestAccKustoCluster_basic (1254.71s)
PASS


Process finished with the exit code 0

**Comment by liuwuliuyun** [comment]
GOROOT=D:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"D:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto #gosetup
"D:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccKustoCluster_complete\E$ #gosetup
=== RUN   TestAccKustoCluster_complete
=== PAUSE TestAccKustoCluster_complete
=== CONT  TestAccKustoCluster_complete
--- PASS: TestAccKustoCluster_complete (1105.06s)
PASS


Process finished with the exit code 0

**Review by katbyte — APPROVED** [review]
LGTM 🧬

**Pull request merged**